flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/CompactManager.java

```diff
@@ -75,7 +75,8 @@ public void submitCompaction(Levels levels) {
                                 return;
                             }
                             boolean dropDelete =
-                                    unit.outputLevel() >= levels.nonEmptyHighestLevel();
+                                    unit.outputLevel() != 0
+                                            && unit.outputLevel() >= levels.nonEmptyHighestLevel();
```

Comment:
add comments to explain the logic (for example level 0 is special because key ranges from different files may overlap, and that compaction are based on runs instead of files)